### PR TITLE
Updates feedback to allow solution views to pass

### PR DESCRIPTION
**Pull request by Hamms**
Fixes [109085120](https://www.pivotaltracker.com/projects/1256652/stories/109085120) and [110114648](https://www.pivotaltracker.com/projects/1256652/stories/110114648)

`getUserBlocks_` is used by feedback to get all the blocks in the BlockSpace that could reasonably be considered to be the blocks provided by the user; namely, enabled, editable blocks that aren't the "when run" block. This list is then checked against the required and recommended blocks to determine if the user can pass.

Unfortunately, in "solution view" mode, Blockly is set to readonly, which makes all blocks noneditable. This means that `getUserBlocks_` returns an empty list, and the solution will always fail the required blocks check. This is causing lots of confusion for teachers.

This PR simply alters `getUserBlocks_` to only filter out noneditable blocks when Blockly is not in readOnly mode.

### CODE REVIEW

**Comment by Hamms** [comment]
Once https://github.com/code-dot-org/code-dot-org/pull/6136 is merged in, this should be updated to look at `Blockly.mainBlockSpace.isReadOnly()` rather than `Blockly.readOnly`

File: apps/test/feedbackTests.js [hunk 1]
```diff
@@ -225,6 +225,56 @@ describe("throwOnInvalidExampleBlocks", function () {
...
+    var testBlockXml = [
+      '<xml>',
+      '<block editable="false" type="text_print"></block>',
+      '<block editable="false" type="text"><title name="TEXT">TextContent</title></block>',
+      '<block editable="false" type="math_number"><title name="NUM">10</title></block>',
+      '</xml>'
+    ];
+
+    var readOnly = Blockly.readOnly;
+    Blockly.readOnly = true;
```
Comment (islemaster) [inline apps/test/feedbackTests.js:271]: Does this test need to be updated?  How does this interact with the block-space-specific readonly (i.e. `Blockly.mainBlockSpace.isReadOnly()`) change?

Comment (Hamms) [inline apps/test/feedbackTests.js:271]: Nope! Blockly.mainBlockSpace.isReadOnly() examines a couple of things to determine readOnly status, including `Blockly.readOnly`

File: apps/test/feedbackTests.js [hunk 2]
```diff
@@ -225,6 +225,56 @@ describe("throwOnInvalidExampleBlocks", function () {
...
+    validateNumUserBlocks(testBlockXml.join(''), 3);
+    Blockly.readOnly = readOnly;
```
Comment (islemaster) [inline apps/test/feedbackTests.js:273]: :( Do we not reinitialize Blockly across tests?  If not, this makes me wish we had a nice pattern for saving/restoring its state (like some graphics APIs do).

**Comment by islemaster** [comment]
LGTM!  Kudos for covering with a test and commenting potentially confusing behavior.

File: apps/test/feedbackTests.js (hunk 1 above)
Comment (Hamms) [inline apps/test/feedbackTests.js:271]: the function itself is [here](https://github.com/code-dot-org/code-dot-org/blob/staging/blockly-core/core/ui/block_space/block_space.js#L208) and the PR that added it is [here](https://github.com/code-dot-org/code-dot-org/pull/6136)

File: apps/test/feedbackTests.js (hunk 2 above)
Comment (Hamms) [inline apps/test/feedbackTests.js:273]: I could probably initialize two Blocklys for this test rather than modifying the one global one.

**Comment by Hamms** [comment]
Thanks!

**Pull request merged**